core/src/main/java/org/apache/accumulo/core/clientImpl/TabletServerBatchWriter.java

```diff
@@ -118,7 +122,7 @@
   // state
   private boolean flushing;
   private boolean closed;
-  private MutationSet mutations;
+  private MutationSet incomingQueue;
```

Comment:
Seems like a BlockingQueue<Mutation> may be better for this, but not sure still looking over the code.  Why was the type MutationsSet chosen?

Reply:
Looking at this a bit more, one thing MutationsSet does is keep track of the table id that goes with a mutation.